hs_file_types/models/base.py

```diff
@@ -612,6 +722,158 @@ def get_allowed_storage_file_types(cls):
     def type_name(cls):
         return cls.__name__
 
+    @classmethod
+    def check_files_for_aggregation_type(cls, files):
+        """Checks if the specified files can be used to set this aggregation type. Sub classes may
+        need to override this.
+        :param  files: a list of ResourceFile objects
+
+        :return If the files meet the requirements of this aggregation type, then returns this
+        aggregation class name, otherwise empty string.
+        """
+        return ""
```

Comment:
Should we raise a NotImplementedError here?

Reply:
Only the aggregation types that support aggregation creation from a folder need to override this. That's why I didn't use raise NotImplementedError. I will update the doc string accordingly.